src/functions/enclose.js

```diff
@@ -0,0 +1,188 @@
+// @flow
+import defineFunction from "../defineFunction";
+import buildCommon from "../buildCommon";
+import mathMLTree from "../mathMLTree";
+import utils from "../utils";
+import stretchy from "../stretchy";
+
+import * as html from "../buildHTML";
+import * as mml from "../buildMathML";
+
+
+const htmlBuilder = (group, options) => {
+    // \cancel, \bcancel, \xcancel, \sout, \fbox, \colorbox, \fcolorbox
+    const inner = html.buildGroup(group.value.body, options);
+
+    const label = group.value.label.substr(1);
+    const scale = options.sizeMultiplier;
+    let img;
+    let imgShift = 0;
+    const isColorbox = /color/.test(label);
+
+    if (label === "sout") {
+        img = buildCommon.makeSpan(["stretchy", "sout"]);
+        img.height = options.fontMetrics().defaultRuleThickness / scale;
+        imgShift = -0.5 * options.fontMetrics().xHeight;
+
+    } else {
+        // Add horizontal padding
+        inner.classes.push(/cancel/.test(label) ? "cancel-pad" : "boxpad");
+
+        // Add vertical padding
+        let vertPad = 0;
+        // ref: LaTeX source2e: \fboxsep = 3pt;  \fboxrule = .4pt
+        // ref: cancel package: \advance\totalheight2\p@ % "+2"
+        if (/box/.test(label)) {
+            vertPad = label === "colorbox" ? 0.3 : 0.34;
+        } else {
+            vertPad = utils.isCharacterBox(group.value.body) ? 0.2 : 0;
+        }
+
+        img = stretchy.encloseSpan(inner, label, vertPad, options);
+        imgShift = inner.depth + vertPad;
+
+        if (isColorbox) {
+            img.style.backgroundColor = group.value.backgroundColor.value;
+            if (label === "fcolorbox") {
+                img.style.borderColor = group.value.borderColor.value;
+            }
+        }
+    }
+
+    let vlist;
+    if (isColorbox) {
+        vlist = buildCommon.makeVList({
+            positionType: "individualShift",
+            children: [
+                // Put the color background behind inner;
+                {type: "elem", elem: img, shift: imgShift},
+                {type: "elem", elem: inner, shift: 0},
+            ],
+        }, options);
+    } else {
+        vlist = buildCommon.makeVList({
+            positionType: "individualShift",
+            children: [
+                // Write the \cancel stroke on top of inner.
+                {
+                    type: "elem",
+                    elem: inner,
+                    shift: 0,
+                },
+                {
+                    type: "elem",
+                    elem: img,
+                    shift: imgShift,
+                    wrapperClasses: /cancel/.test(label) ? ["svg-align"] : [],
```

Comment:
We were doing `vlist.children[0].children[0].children[1].classes.push("svg-align");` before, but flow didn't like it because not all `children` have great grand children.  This specifies classes to be added to the wrapper spans up front instead of afterwards.

Reply:
Not to mention the original code is much harder to reason about. This quite an improvement.